src/control/server/storage/bdev.go

```diff
@@ -24,6 +31,61 @@ import (
 // BdevPciAddrSep defines the separator used between PCI addresses in string lists.
 const BdevPciAddrSep = " "
 
+// BioState represents the health state of BIO device on an NVMe device (a blobstore on a NVMe
+// namespace).
+type BioState uint32
```

Comment:
Unless I'm missing something, I don't see how this can work correctly to distinctly represent both nvme and bio states. On the C side, for example, NVME_DEV_FL_PLUGGED is 1, and BIO_DEV_FAULTY is also 1.

I think you're going to need separate bitmasks or you're going to need some kind of offset scheme to shift the values for one or the other enum set. It would be much clearer and maintainable to have two separate bitmasks though, I think.

Reply:
NVME_DEV_FL has now encompassed BIO_DEV (we now use only one set of NVME_DEV_FL bit flags, BIO_DEV enum has been removed).

Reply:
I retained the BioState nomenclature on the control side as that seems relevant to it's perspective (the control plane is ingesting device state from BIO).

Reply:
Ah, my mistake. I missed that this PR was also removing the bio_dev_state enum.

Still, "BioState" seems a bit imprecise and implies that that the bitmask holds information about Bio, whatever that is. Particularly when read in a context where the type will be used outside of the storage package, e.g. `storage.BioState`. Wouldn't it be clearer to name this something like `storage.NvmeDevState` ? It's a bit more verbose but it's very clear.

Reply:
do you also want to rename proto definitions appropriately or do you just want naming updates in go code?

Reply:
Since you have to modify the type anyhow it seems fine to also rename it.